src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -608,6 +610,97 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
       }
     }
 
+  def queryEntities(workspaceName: WorkspaceName, entityType: String, requestedQuery: EntityQuery): Future[PerRequestMessage] = {
+    object AttributeStringifier {
+      def apply(attribute: Attribute): String = {
+        attribute match {
+          case AttributeNull => ""
+          case AttributeString(value) => value
+          case AttributeNumber(value) => value.toString()
+          case AttributeBoolean(value) => value.toString()
+          case AttributeEmptyList => ""
+          case AttributeValueList(list) => list.map(apply).mkString(" ")
+          case AttributeEntityReferenceList(list) => list.map(apply).mkString(" ")
+          case AttributeEntityReference(t, name) => name
+        }
+      }
+    }
+
+    object AttributeOrderingAsc extends Ordering[Attribute] {
+      override def compare(x: Attribute, y: Attribute): Int = {
+        (x, y) match {
+          case (AttributeNumber(value1), AttributeNumber(value2)) => value1.compare(value2)
+          case (_, _) => AttributeStringifier(x).compareToIgnoreCase(AttributeStringifier(y))
```

Comment:
So this'll give us:

* `true` and `false` in the middle of the Ts and Fs
* lists wherever the first letter of their first element puts them
* lists of numbers will become alpha-ordered rather than numerically-ordered by their first element

This is more an observation than anything else. I think the last one is a bit egregious, and if we're special-casing numbers we should special-case lists-of-numbers to be consistent. As for the others, we *could* have a conversation and agree on an ordering of `null > bool > number > string > ref > list` but that's probably excessive.